controllers/replicationsource_controller.go

```diff
@@ -139,7 +143,9 @@ func reconcileSrcUsingCatalog(
 	// Search the Mover catalog for a suitable data mover
 	var dataMover mover.Mover
 	for _, builder := range mover.Catalog {
-		if candidate, err := builder.FromSource(sr.Client, logger, instance); err == nil && candidate != nil {
+		candidate, err := builder.FromSource(sr.Client, logger,
+			record.NewEventRecorderAdapter(sr.EventRecorder), instance)
```

Comment:
@JohnStrunk Just for my knowledge, what is the purpose of wrapping with NewEventRecorderAdapter()?

Reply:
This is the "new" vs "old" Event interface. The new interface is [`events.EventRecorder`](https://pkg.go.dev/k8s.io/client-go@v0.23.5/tools/events#EventRecorder), and the old one is [`record.EventRecorder`](https://pkg.go.dev/k8s.io/client-go@v0.23.5/tools/record#EventRecorder). The new interface publishes events.k8s.io/Event objects whereas the old one is v1/Event. The k8s.io version adds a couple of additional fields for "related" object and "action". Both event types look the same when doing a `kubectl describe <object>`.

The adapter allows us to use the new interface and provide those additional fields even though controller-runtime is still using the old interface. Those extra fields will be dropped by the adapter, though we could provide our own adapter to tack them onto the message if we wanted. I had written a custom adapter that added `related` & `action` as k/v pairs to the `message` field, but I decided to drop it from this PR since it wasn't really providing any useful information.

Once controller-runtime includes the new interface, we'll drop the adapter.